From b20e91f335efb83550ae66bd194a18df7c6f4ee5 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Thu, 31 Mar 2022 11:21:19 +0200
Subject: [PATCH 43/54] fix(LinstorVolumeManager): ensure we always use IPs in
 _get_controller_uri

Otherwise if a hostname is returned, we can't use it if the XCP-ng pool
is configurer using static IPs instead of DHCP.

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/linstorvolumemanager.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/linstorvolumemanager.py b/drivers/linstorvolumemanager.py
index b4ee783..2d5c63e 100755
--- a/drivers/linstorvolumemanager.py
+++ b/drivers/linstorvolumemanager.py
@@ -192,7 +192,7 @@ def _get_controller_uri():
             if distutils.util.strtobool(
                 session.xenapi.host.call_plugin(host_ref, PLUGIN, PLUGIN_CMD, {})
             ):
-                return 'linstor://' + host_record['hostname']
+                return 'linstor://' + host_record['address']
     except:
         # Not found, maybe we are trying to create the SR...
         pass
-- 
2.36.0

